js-sdk/src/googlePay/GooglePayButtonWidget.ts

```diff
@@ -0,0 +1,176 @@
+import { PublicSquare } from '@/PublicSquare';
+import type { GooglePayButtonWidgetOptions, GooglePayConfiguration, PublicSquareInitOptions } from '@/types'
+
+export class GooglePayButtonWidget {
+  private options: Required<GooglePayButtonWidgetOptions>;
+  private containerRef: HTMLElement | null;
+  private paymentsClient: any;
+  private publicSquare = new PublicSquare();
+  private baseRequest = {
+    apiVersion: 2,
+    apiVersionMinor: 0,
+  };
+
+  constructor(options: GooglePayButtonWidgetOptions = {}) {
+    this.options = {
+      id: options.id || 'google-pay-btn',
+      environment: options.environment || 'TEST',
+      merchantId: options.merchantId || '12345678901234567890',
+      merchantName: options.merchantName || 'Example Merchant',
+      allowedCardAuthMethods: options.allowedCardAuthMethods || ['PAN_ONLY', 'CRYPTOGRAM_3DS'],
+      buttonColor: options.buttonColor || 'black',
+      buttonType: options.buttonType || 'buy',
+      locale: options.locale || 'en',
+      style: options.style || { width: '160px', height: '40px', borderRadius: 4, borderType: 'default_border' },
+      transactionInfo: options.transactionInfo || {
```

Comment:
I don't think we want to default this. We should probably check that it was set.